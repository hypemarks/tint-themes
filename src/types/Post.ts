import { IPost, Nullable } from '@tintup/tint-sdk';

export interface ICustomCTA {
  text: string;
  url: string;
  thumbnail_url: Nullable<string>;
  type: string;
  coordinates: Nullable<string[]>;
  id: string;
}

export interface IPostWithCTA extends IPost {
  cta: ICustomCTA[];
}
