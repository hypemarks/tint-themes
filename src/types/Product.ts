import { IProductTags, ITaggedProduct } from '@tintup/tint-sdk/lib';

export { IProductTags, ITaggedProduct };
