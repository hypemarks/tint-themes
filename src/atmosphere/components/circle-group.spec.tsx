import React from 'react';
import { render } from '@testing-library/react';
import CircleGroup from './circle-group';
import { buildPost } from '../../../test/generate';
import { IPostWithCTA } from '../../types/Post';

jest.mock('react', () => {
  const originReact = jest.requireActual('react');
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef,
  };
});
describe('Test CircleGroup Component', () => {
  it('should render snapshot', () => {
    const wrapper = render(
      <svg>
        <CircleGroup
          posts={[buildPost() as IPostWithCTA]}
          atmosphereBlipColors={[]}
          radius={23}
          xCenterPosition={2}
          yCenterPosition={3}
          circleGroupRef={jest.fn(() => ({})) as any}
        />
      </svg>
    );
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
