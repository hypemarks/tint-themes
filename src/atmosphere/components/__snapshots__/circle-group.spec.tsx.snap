// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CircleGroup Component should render snapshot 1`] = `
<DocumentFragment>
  <svg>
    <group
      rotation="45"
      x="2"
      y="3"
    >
      <circle
        position="[object Object]"
        radius="4"
      />
    </group>
  </svg>
</DocumentFragment>
`;
